contracts/Tokenomics.sol

```diff
@@ -1313,55 +1325,52 @@ contract Tokenomics is TokenomicsConstants {
-        uint256 curEffectiveBond = effectiveBond;
-        uint256 curMaxBond = maxBond;
-        if (curMaxBond > curEffectiveBond) {
-            revert Overflow(curMaxBond, curEffectiveBond);
-        }
-
-        // Adjust effective bond by reducing it with the on-going epoch maxBond value, resetting current epoch maxBond
-        curEffectiveBond -= curMaxBond;
-
-        // Recalculate inflation per second based on the updated current year inflation
-        // If the inflation curve has not changed, the inflation per second value will be the same as before
-        // Get current inflation per second
-        uint256 curInflationPerSecond = inflationPerSecond;
         // Get last epoch end time
         uint256 curEpochCounter = epochCounter;
         uint256 endTime = mapEpochTokenomics[curEpochCounter - 1].epochPoint.endTime;
-        // Calculate seconds passed in a year until last epoch end time
-        uint256 secondsInYearUntilEndTime = endTime - numYears * ONE_YEAR - timeLaunch;
-        // Get old inflation from beginning of the year until end of last epoch
-        uint256 oldInflationUntilEpochEnd = curInflationPerSecond * secondsInYearUntilEndTime;
-        // Calculate updated inflation per second
-        curInflationPerSecond = getInflationForYear(currentYear) / ONE_YEAR;
-        // Get new inflation from beginning of the year until end of last epoch
-        uint256 newInflationUntilEpochEnd = curInflationPerSecond * secondsInYearUntilEndTime;
-        // Check for inflation reduction
-        if (newInflationUntilEpochEnd > oldInflationUntilEpochEnd) {
-            revert Overflow(newInflationUntilEpochEnd, oldInflationUntilEpochEnd);
+        // Find if this epoch is the year change epoch
+        uint256 lastEpochYear = (endTime - timeLaunch) / ONE_YEAR;
+        if (numYears > lastEpochYear) {
+            revert Overflow(numYears, lastEpochYear);
+        }
+
+        // Reset staking inflation
+        // Assume there are no more than 100 epochs to clear retained staking amount
+        for (uint256 i = 0; i < 100; ++i) {
+            // This is a low level call since it must never revert
+            (bool success, ) = dispenser.call(abi.encodeCall(IDispenser.retain, ()));
+
+            // Break when all the retains are executed
+            if (!success) {
+                break;
+            }
         }
+        mapEpochStakingPoints[curEpochCounter].stakingIncentive = 0;
 
-        // Calculate inflation reduction
-        uint256 reductionInflation = oldInflationUntilEpochEnd - newInflationUntilEpochEnd;
-        // Check for effective bond cut in favor of inflation difference
-        if (reductionInflation > curEffectiveBond) {
-            revert Overflow(reductionInflation, curEffectiveBond);
+        // Update top-up fractions
+        uint256 sumTopUpFractions = _maxBondFraction + _topUpComponentFraction + _topUpAgentFraction + _stakingFraction;
```

Comment:
Maybe let's rename this with sum fractions and let's update the comment above accordingly